### PR TITLE
Update ChannelHeaderPlug tests to not use snapshots and ensure its more accessible

### PR DESCRIPTION
**Pull request by hmhealey**
#### Summary
I made these changes a few weeks ago as part of switching this component to `WithTooltip`. Because that broke the tests for that component, I also went and migrated the tests to Testing Library and I got rid of the usage of `localizeMessage` in it as well which required making it more accessible too

#### Release Note
```release-note
Improve accessibility of plugin buttons in the channel header
```

### CODE REVIEW

File: webapp/channels/src/components/channel_header/components/header_icon_wrapper.tsx [hunk 1]
```diff
@@ -2,198 +2,72 @@
...
-                }
-            </div>
-        );
-    }
+    const ariaLabelText = ariaLabelOverride ?? tooltipText;
 
     return (
-        <>
-            <div className='flex-child'>
+        <div>
```
[the remark this answers is not in the document]
Comment (hmhealey) [reply webapp/channels/src/components/channel_header/components/header_icon_wrapper.tsx:48]: No, there isn't. It's left over from when I was fighting with something while working on this. I'll remove it

**Review by cpoile — APPROVED** [review]
Thanks Harrison!

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**